delta-app/src/main/java/io/cdap/delta/store/DraftService.java

```diff
@@ -206,10 +206,18 @@ public TableAssessmentResponse assessTable(DraftId draftId, Configurer configure
     if (target == null) {
       throw new InvalidDraftException("Cannot assess a table without a configured target.");
     }
+    SourceTable selectedTable = deltaConfig.getTables()
+            .stream()
+            .filter(streamTable -> db.equals(streamTable.getDatabase()) && table.equals(streamTable.getTable()))
+            .findAny()
+            .orElse(null);
+    if (selectedTable == null) {
+      throw new TableNotFoundException(db, table, "Cannot assess table.");
```

Comment:
Doesn't seem like we should hit this case, but should use a better error message anyway. Something like:

table 'xyz' in db 'abc' is not a selected table in the draft.

This also shouldn't be a TableNotFoundException, as that exception is used to denote that the table doesn't exist in the database (some sort of user error). This is more like an IllegalArgumentException, as it indicates we have a bug somewhere in our logic.